pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java

```diff
@@ -239,6 +239,7 @@ private static TransferableBlock composeSelectTransferableBlock(InstanceResponse
           + " Column Ordering: " + Arrays.toString(columnIndices));
       return composeColumnIndexedTransferableBlock(responseBlock, adjustedResultSchema, columnIndices);
     } else {
+      handleBytesDataType(desiredDataSchema.getColumnDataTypes(), resultSchema.getColumnDataTypes());
```

Comment:
handling this here is wrong IMO. b/c both column index reordering and direct transferrable had to handle bytes type. you can add the logic in `canonicalizeRow` method IMO

Reply:
It is updated as per the comments. Please have another look. Thanks